packages/edit-site/src/components/add-new-pattern/index.js

```diff
@@ -114,8 +128,16 @@ export default function AddNewPattern() {
 						showTooltip: false,
 						__next40pxDefaultSize: true,
 					} }
-					text={ addNewPatternLabel }
-					label={ addNewPatternLabel }
+					text={
+						canCreatePattern
+							? addNewPatternLabel
+							: addNewTemplatePartLabel
+					}
+					label={
+						canCreatePattern
+							? addNewPatternLabel
+							: addNewTemplatePartLabel
```

Comment:
Why are we switching here now? What if you can do neither?

Reply:
It felt strange to have a menu saying Add New Pattern, while adding a template part was the option. If the user can not add template parts and can not add patterns, the condition if ( controls.length === 0 ) { will be true and we will not render any menu.

Reply:
There's also a check for `addNewPatternLabel` which is maybe a bit strange now.

Reply:
Hi @ellatrix, I reverted this change and now we just focus on checking permissions before rendering controls.